src/NzbDrone.Core/Update/InstallUpdateService.cs

```diff
@@ -108,13 +119,35 @@ private void InstallUpdate(UpdatePackage updatePackage)
                                             updateSandboxFolder);
 
                 _logger.Info("Starting update client {0}", _appFolderInfo.GetUpdateClientExePath());
-                _logger.ProgressInfo("NzbDrone will restart shortly.");
+                _logger.ProgressInfo("Sonarr will restart shortly.");
 
                 _processProvider.Start(_appFolderInfo.GetUpdateClientExePath(), GetUpdaterArgs(updateSandboxFolder));
+
+                return;
             }
-            catch (Exception ex)
+            catch (UpdateFailedException ex)
```

Comment:
your second paragraph sounds like two component that shouldn't know about each other know why to much. this method is catching a specific exception because you know it get swallowed somewhere else.